cerk/src/kernel/kernel_start.rs

```diff
@@ -8,22 +8,57 @@ use crate::kernel::{CloudEventMessageRoutingId, ProcessingResult};
 use crate::runtime::channel::{BoxedReceiver, BoxedSender};
 use crate::runtime::InternalServerId;
 use std::collections::HashMap;
+use std::ops::Add;
+use std::sync::Mutex;
+use std::time::{Duration, SystemTime};
 
 const ROUTER_ID: &str = "router";
 const CONFIG_LOADER_ID: &str = "config_loader";
+const ROUTING_TTL_MS: u64 = 100;
 
 struct PendingDelivery {
     sender: InternalServerId,
     missing_receivers: Vec<InternalServerId>,
+    ttl: SystemTime,
 }
 
 type Outboxes = HashMap<InternalServerId, BoxedSender>;
 type PendingDeliveries = HashMap<CloudEventMessageRoutingId, PendingDelivery>;
+type PendingDeliveriesMutex = Mutex<PendingDeliveries>;
```

Comment:
why do we need a mutex. PendingDeliveries are only used in the kernel thread, right?